packages/react-components/src/components/InputBoxComponent.tsx

```diff
@@ -201,3 +634,818 @@ export const InputBoxButton = (props: InputBoxButtonProps): JSX.Element => {
+/**
+ * Find mention tag if selection is inside of it
+ *
+ * @private
+ */
+const findMentionTagForSelection = (tags: TagData[], selection: number): TagData | undefined => {
+  let mentionTag: TagData | undefined = undefined;
+  for (let i = 0; i < tags.length; i++) {
+    const tag = tags[i];
+    let plainTextEndIndex = 0;
+    if (tag.plainTextEndIndex !== undefined && tag.closeTagIdx !== undefined) {
+      // close tag exists
+      plainTextEndIndex = tag.plainTextEndIndex;
+    } else if (tag.plainTextBeginIndex !== undefined) {
+      //no close tag
+      plainTextEndIndex = tag.plainTextBeginIndex;
+    }
+    if (tag.subTags !== undefined && tag.subTags.length !== 0) {
+      const selectedTag = findMentionTagForSelection(tag.subTags, selection);
+      if (selectedTag !== undefined) {
+        mentionTag = selectedTag;
+        break;
+      }
+    } else if (
+      tag.tagType === 'msft-mention' &&
+      tag.plainTextBeginIndex !== undefined &&
+      tag.plainTextBeginIndex < selection &&
+      selection < plainTextEndIndex
+    ) {
+      mentionTag = tag;
+      break;
+    }
+  }
+  return mentionTag;
+};
+
+/* @conditional-compile-remove(mention) */
+/**
+ * Go through tags and find a new the selection index if it is inside of a mention tag
+ *
+ * @private
+ */
+const findNewSelectionIndexForMention = (
+  tag: TagData,
+  textValue: string,
+  selection: number,
+  previousSelection: number
+): number => {
+  if (
+    tag.plainTextBeginIndex === undefined ||
+    tag.tagType !== 'msft-mention' ||
+    selection === previousSelection ||
+    tag.plainTextEndIndex === undefined
+  ) {
+    return selection;
+  }
+  let spaceIndex = 0;
+  if (selection <= previousSelection) {
+    // the cursor is moved to the left
+    spaceIndex = textValue.lastIndexOf(' ', selection ?? 0);
+    if (spaceIndex === -1) {
+      // no space before the selection
+      spaceIndex = tag.plainTextBeginIndex;
+    }
+  } else {
+    // the cursor is moved to the right
+    spaceIndex = textValue.indexOf(' ', selection ?? 0);
+    if (spaceIndex === -1) {
+      // no space after the selection
+      spaceIndex = tag.plainTextEndIndex ?? tag.plainTextBeginIndex;
+    }
+  }
+
+  if (spaceIndex < tag.plainTextBeginIndex) {
+    spaceIndex = tag.plainTextBeginIndex;
+  } else if (spaceIndex > tag.plainTextEndIndex) {
+    spaceIndex = tag.plainTextEndIndex;
+  }
+  return spaceIndex;
+};
+
+/* @conditional-compile-remove(mention) */
+/**
+ * Handle mention tag edit and by word deleting
+ *
+ * @private
+ */
+const handleMentionTagUpdate = (
+  htmlText: string,
+  oldPlainText: string,
+  lastProcessedHTMLIndex: number,
+  processedChange: string,
+  tag: TagData,
+  closeTagIdx: number,
+  closeTagLength: number,
+  plainTextEndIndex: number,
+  startIndex: number,
+  oldPlainTextEndIndex: number,
+  mentionTagLength: number
+): [resultValue: string, updatedChange: string, htmlIndex: number, plainTextSelectionEndIndex: number | null] => {
```

Comment:
I am less concerned about private implementation, but we do prefer returning object instead of tuple, it's hard to read from the caller part

Reply:
We will address in the refactor task; I agree.